demo_nodes_py/README.md

```diff
@@ -0,0 +1,280 @@
+## **What Is This?**
+
+This demo ...
+
+This package consists of ROS 2 nodes:
+
+1. `listener`
+2. `talker`
+3. `add_two_ints_client`
+4. `add_two_ints_server`
+5. `set_parameters_callback`
+6. `add_two_ints_client_async`
+7. `listener_qos`
+8. `talker_qos`
+9. `listerner_serialized`
+10. `async_param_client`
+
+## **Build**
+
+Run this command to build the package:
+
+```bash
+colcon build --packages-select demo_nodes_py
+```
+
+## **Run**
+
+### Basic Listener & Talker
+
+This runs `talker` and `listener` ROS 2 nodes which exchanges the following string with incremeting integer:
+
+> Hello World: <count_>
+
+```bash
+# Open new terminal
+ros2 run demo_nodes_py talker
+```
+
+```bash
+# Open new terminal
+ros2 run demo_nodes_py listener
+```
+
+![](img/talker_listener.png)
+
+### Server & Client
+
+This runs `add_two_ints_client` and `add_two_ints_server` ROS 2 client and server where  Server processes two integers sent from Client and publishes its sum to Client to be printed out.
+
+#### Server
+
+```bash
+# Open new terminal
+ros2 run demo_nodes_py add_two_ints_server
+```
+
+#### Client [Synchronous]
+
+```bash
+# Open new terminal
+ros2 run demo_nodes_py add_two_ints_client
+```
+
+#### Client [Asynchronous]
+
+```bash
+# Open new terminal
+ros2 run demo_nodes_py add_two_ints_client_async
+```
+![](img/server_client.png)
+
+### QoS Listener & Talker
+
+Similar to previous, this runs `talker_qos` and `listener_qos` ROS 2 nodes which exchanges using the Quality of Service (QoS) Best Effort policy.
```

Comment:
"... which exchange messages using the Quality of Service (QoS) best effort policy."

Reply:
Done.